ivy_tests/test_ivy/test_functional/test_core/test_gradients.py

```diff
@@ -4,164 +4,196 @@
 from numbers import Number
 import pytest
 import numpy as np
+import torch
+import tensorflow as tf
+from hypothesis import given, strategies as st
 
 # local
 import ivy
 import ivy.functional.backends.numpy
+from ivy.functional.ivy.creation import native_array
 import ivy_tests.test_ivy.helpers as helpers
 from ivy.container import Container
+import ivy.functional.backends.numpy as ivy_np
 
 
-# variable
-@pytest.mark.parametrize("object_in", [[], [0.0], [1], [True], [[1.0, 2.0]]])
-@pytest.mark.parametrize("dtype", ["float16", "float32", "float64"])
-def test_variable(object_in, dtype, device, call):
-    if call is helpers.tf_graph_call:
-        # cannot create variables as part of compiled tf graph
-        pytest.skip()
-    if call in [helpers.mx_call] and dtype == "int16":
-        # mxnet does not support int16
-        pytest.skip()
-    if len(object_in) == 0 and call is helpers.mx_call:
-        # mxnet does not support 0-dimensional variables
-        pytest.skip()
-    # smoke test
-    ret = ivy.variable(ivy.array(object_in, dtype=dtype, device=device))
-    # type test
-    if call is not helpers.np_call:
-        assert ivy.is_variable(ret)
-    # cardinality test
-    assert ret.shape == np.array(object_in).shape
-    # value test
-    assert np.allclose(
-        call(ivy.variable, ivy.array(object_in, dtype=dtype, device=device)),
-        np.array(object_in).astype(dtype),
-    )
-    # compilation test
-    if call in [helpers.torch_call]:
-        # pytorch scripting does not support string devices
+# variable   
+@given(
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="variable"),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_variable(
+    dtype_and_x,
+    as_variable,
+    with_out,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw
+):
+    dtype, x = dtype_and_x
+    x = np.asarray(x, dtype=dtype)
+    if x.shape == ():
```

Comment:
we shouldn't check for certain shapes in the test body itself, we should instead avoid generating these shapes in the first place, by extending the hypothesis helper functions, or passing in different arguments.